plugin/mutate/source/dextool/plugin/mutate/backend/analyze/package.d

```diff
@@ -1115,17 +1127,48 @@ struct Analyze {
         if (auto localId = file in result.idFile) {
             const fid = FileId(localId.get);
 
+            int sectionStart = -1;
             auto mdata = appender!(LineMetadata[])();
             foreach (t; tstream.getTokens(file).filter!(a => a.kind == CXTokenKind.comment)) {
-                auto m = matchFirst(t.spelling, re_nomut);
-                if (m.whichPattern == 0)
-                    continue;
-
-                () @trusted {
-                    mdata.put(LineMetadata(fid, t.loc.line,
-                            LineAttr(NoMut(m["tag"], m["comment"]))));
-                }();
-                log.tracef("NOMUT found at %s:%s:%s", file, t.loc.line, t.loc.column);
+                auto m1 = matchFirst(t.spelling, re_nomut);
+                auto m2 = matchFirst(t.spelling, re_nomut_begin);
+                auto m3 = matchFirst(t.spelling, re_nomut_end);
+                auto m4 = matchFirst(t.spelling, re_nomut_next); 
+
+                if (m1.whichPattern != 0 && m2.whichPattern == 0 && m3.whichPattern == 0 && m4.whichPattern == 0){
```

Comment:
if the change to the regex works then this code basically becomes a switch-statement with cases for "BEGIN", "END", "NEXT" and default (empty).